internal/pkg/agent/application/inject_proxy_component_modifier.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"os"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator"
+	"github.com/elastic/elastic-agent/pkg/component"
+)
+
+// InjectProxyComponentModifier injects a proxy_url value into a component's config if one is not set.
+//
+// This will inject a proxy_url into each output unit if there is no existing value and the agent has a proxy set through an environment variable.
+// The URL used is the HTTPS_PROXY env var. If that's not set the HTTP_PROXY env var is used.
+// If there are no env vars set, or the unit's config has `proxy_disable: true`, nothing is injected
+func InjectProxyComponentModifier() coordinator.ComponentsModifier {
+	proxyURL, ok := os.LookupEnv("HTTPS_PROXY")
+	if !ok {
+		proxyURL, _ = os.LookupEnv("HTTP_PROXY")
+	}
```

Comment:
This means that the `HTTPS_PROXY` will take over the `HTTP_PROXY` even in the case that `HTTPS_PROXY` is set to `""`. That might not be the behavior wanted, might be better to use:

```suggestion
	proxyURL := os.Getenv("HTTPS_PROXY")
	if proxyURL == "" {
		proxyURL = os.LookupEnv("HTTP_PROXY")
	}
```